### ISSUE TITLE
segfault trying to memcpy

### ISSUE THREAD

**Issue by andrewclink** [achilikin/bdfe#2]
I got this to build on OS X by stubbing out the I2C drivers (which would be awesome, but alas).
However, after generating  few characters I'm getting a segfault at ~bdf.c:259 which is:
`memcpy(gout, gin, dy);`

It seems that gin gets corrupted somewhere (gdb reports it's "out of bounds"). 

Debugging the pointers shows, for example:

```
memcpy(0x10080360c, 0x100103765, 16)
memcpy(0x10080361c, 0x100103770, 16)
memcpy(0x10080362c, 0x100103770, 16)
memcpy(0x10080363c, 0x100103770, 16)
memcpy(0x10080364c, 0x103771, 16)
< segfault >
```

I'm using a BFD file I generated. Maybe you could post one that's known working?

**Comment by andrewclink** [comment]
The pointer gets corrupted by the line
`gin -= displacement + ascender`

In this case, `displacement` == 0xFFFFFFFF. Changing its type to `int` instead of `unsigned` fixes the crash for me. I don't know the code as well as you do, so I don't know if it's more appropriate to guard against unreasonably large `displacement` values, or change it to signed (and guard against negative values?).

Anyways, hope this helps.

**Comment by achilikin** [comment]
I have not seen any fonts with displacement going negative, but no harm to use int instead of unsigned.

**Comment by achilikin** [comment]
I've changed unsifned to integer in the latest commit.
Could you please give me you file which cases negative displacement? Just want to trace and see why it is happening.

**Comment by andrewclink** [comment]
Unfortunately I don't recall which files I used that crashed. I hacked up my copy trying to support large font sizes, and ultimately went a different direction. I switched the declaration back to `unsigned` but I can't reproduce the crash anymore.

I suspect it was a larger point-size than supported that generated the crash; my application is a watch that benefits from a nearly 64-pixel-tall font. One is attached, but not sure if it'll help or not.

[system_big.bdf.zip](https://github.com/achilikin/bdfe/files/517755/system_big.bdf.zip)
